qiskit/scheduler/methods/basic.py

```diff
@@ -0,0 +1,180 @@
+        """Update the time tracker for all inst_qubits to the given time."""
+        for q in inst_qubits:
+            # A newly scheduled instruction on q starts at t=0 as we move backwards
+            qubit_available_until[q] = 0
+        for q in qubit_available_until.keys():
+            if q not in inst_qubits:
+                # Uninvolved qubits might be free for the duration of the new instruction
+                qubit_available_until[q] += shift
+
+    circ_pulse_defs = translate_gates_to_pulse_defs(circuit, schedule_config)
+    for circ_pulse_def in reversed(circ_pulse_defs):
+        if isinstance(circ_pulse_def.schedule, Barrier):
+            update_times(circ_pulse_def.qubits)
+        else:
+            cmd_sched = circ_pulse_def.schedule
+            # The new instruction should end when one of its qubits becomes occupied
+            cmd_start_time = (min([qubit_available_until[q] for q in circ_pulse_def.qubits])
+                              - cmd_sched.duration)
+            if cmd_start_time == float("inf"):
+                # These qubits haven't been used yet, so schedule the instruction at t=0
+                cmd_start_time = 0
+            # We have to translate qubit times forward when the cmd_start_time is negative
+            shift_amount = max(0, -cmd_start_time)
+            cmd_start_time = max(cmd_start_time, 0)
+            sched = cmd_sched.shift(cmd_start_time).insert(shift_amount, sched, name=sched.name)
+            update_times(circ_pulse_def.qubits, shift_amount)
+    return sched
+
+
+def translate_gates_to_pulse_defs(circuit: QuantumCircuit,
+                                  schedule_config: ScheduleConfig) -> List[CircuitPulseDef]:
+    """
+    Without concern for the final schedule, extract and return a list of Schedules and the qubits
+    they operate on, for each element encountered in the input circuit. Measures are grouped when
+    possible, so qc.measure(q0, c0)/qc.measure(q1, c1) will generate a synchronous measurement
+    pulse.
+
+    Args:
+        circuit: The quantum circuit to translate
+        schedule_config: Backend specific parameters used for building the Schedule
+    Returns:
+        A list of CircuitPulseDefs: the pulse definition for each circuit element
+    Raises:
+        QiskitError: If circuit uses a command that isn't defined in config.cmd_def
+    """
+    circ_pulse_defs = []
+
+    cmd_def = schedule_config.cmd_def
+    measured_qubits = set()  # Collect qubits that would like to be measured
+
+    def get_measure_schedule() -> CircuitPulseDef:
+        """Create a schedule to measure the qubits queued for measuring."""
+        measures = set()
+        all_qubits = set()
+        sched = Schedule()
+        for q in measured_qubits:
+            measures.add(tuple(schedule_config.meas_map[q]))
+        for qubits in measures:
+            all_qubits.update(qubits)
+            # TODO (Issue #2704): Respect MemorySlots from the input circuit
```

Comment:
This could be resolved by putting the non-requested qubits in ancilla `memory_slots`.